pkg/datastore/commandstore.go

```diff
@@ -86,6 +86,35 @@ func (c *commandCollection) Decode(e interface{}, parts ...[]byte) error {
 	return nil
 }
 
+func (c *commandCollection) Encode(e interface{}) (map[Shard][]byte, error) {
+	errFmt := "failed while encode Command object: %s"
+
+	me, ok := e.(*model.Command)
+	if !ok {
+		return nil, fmt.Errorf(errFmt, "type not matched")
+	}
+
+	agentShardStruct := me
+	adata, err := json.Marshal(&agentShardStruct)
```

Comment:
Is it to improve more readability? if so, it's okay.
```suggestion
	adata, err := json.Marshal(me)
```

Reply:
yes, it's 👍 lets me keep it as is to make it more clear that we're building data for agent shard 😄

Reply:
I got you. Pretty good!